### DISCUSSION TITLE
分账签名算法问题

### DISCUSSION THREAD

**Discussion by 54481280** [w7corp/easywechat#2534] [Ideas]
![image](https://user-images.githubusercontent.com/45725624/177547991-2784b8bc-f726-4a17-b81c-c6e109477643.png)
![image](https://user-images.githubusercontent.com/45725624/177548021-5325a351-bf9c-4009-b5c6-6f0434339d45.png)

**Comment by 54481280** [comment]
还望官方修复

**Comment by overtrue** [comment]
@54481280 
1. 什么版本？
2. 什么问题？

**Reply by 54481280** [reply]
4.0，调用分账相关接口时，一直签名错误，最终排查到算法不一样，源码使用的是md5加密，官方给的是HMAC-SHA256加密

**Comment by TheNorthMemory** [comment]
自己造个 `$encryptMethod` 就可以。

```php
$encryptMethod = function (string $thing) {
    \parse_str($thing, $things);
    ['key' => $key] = $things;

    return \hash_hmac('sha256', $thing, $key);
};
```

另外，截图贴代码，这习惯贼差，图片不能参与检索、无法引用，建议改掉这个不好得习惯。
